jinahub/encoders/image/CLIPImageEncoder/clip_image.py

```diff
@@ -1,123 +1,122 @@
-            be an ``ndarray`` with ``dtype=uint8`` or ``dtype=float32``.
-            The ``Height`` and ``Width`` can have arbitrary values.
-
-            If you set ``use_default_preprocessing=True`` when creating this encoder,
-            then the image arrays should have the shape ``[H, W, 3]``, and be in the
-            RGB color format.
-
-            If you set ``use_default_preprocessing=False`` when creating this encoder,
-            then you need to ensure that the images you pass in are already
-            pre-processed. This means that they are all the same size (for batching) -
-            the CLIP model was trained on ``224 x 224`` images, and that they are of
-            the shape ``[3 H, W]`` (in the RGB color format). They should also be
-            normalized.
-        :param parameters: A dictionary that contains parameters to control encoding.
-            The accepted keys are ``traversal_paths`` and ``batch_size`` - in their
-            absence their corresponding default values are used.
-        """
-        if docs:
-            document_batches_generator = get_docs_batch_generator(
-                docs,
-                traversal_path=parameters.get(
-                    "traversal_paths", self.default_traversal_paths
-                ),
-                batch_size=parameters.get("batch_size", self.default_batch_size),
-                needs_attr="blob",
-            )
-
-            with torch.no_grad():
-                for batch_docs in document_batches_generator:
-                    blob_batch = [d.blob for d in batch_docs]
-                    if self.use_default_preprocessing:
-                        tensor = self._generate_input_features(blob_batch.copy())
-                    else:
-                        tensor = {
-                            "pixel_values": torch.tensor(
-                                blob_batch.copy(),
-                                dtype=torch.float32,
-                                device=self.device,
-                            )
-                        }
-
-                    embeddings = self.model.get_image_features(**tensor)
-                    embeddings = embeddings.cpu().numpy()
-
-                    for doc, embed in zip(batch_docs, embeddings):
-                        doc.embedding = embed
-
-    def _generate_input_features(self, images):
-        input_tokens = self.preprocessor(
-            images=images,
-            return_tensors="pt",
-        )
-        input_tokens = {
-            k: v.to(torch.device(self.device)) for k, v in input_tokens.items()
-        }
-        return input_tokens
+from typing import Optional, Tuple
+
+import torch
+from jina import DocumentArray, Executor, requests
+from jina.logging.logger import JinaLogger
+from jina_commons.batching import get_docs_batch_generator
+from transformers import CLIPFeatureExtractor, CLIPModel
+
+
+class CLIPImageEncoder(Executor):
```

Comment:
The diff for this file looks odd.
![Screenshot from 2021-09-16 15-41-53](https://user-images.githubusercontent.com/8330330/133622916-1839cc0f-072e-427e-a5ab-90681ec9eda3.png)

Reply:
according to pycharm, the difference is the new line character (one is CRLF and the other is LF)
![image](https://user-images.githubusercontent.com/15269265/133633189-b648fa57-5830-41d1-81ef-4bed1ac5b6ba.png)
This appeared when I merged main to the branch:
https://github.com/jina-ai/executors/pull/203/commits/bc4422d29faaea4354193192d3996b113875cfdb

Reply:
I suggest we keep it as it is since LF is the new line character in linux
maybe this happened when someone worked on a different OS